assets/app/scripts/services/routes.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("RoutesService", function() {
+    var isPortNamed = function(port) {
+      return angular.isString(port);
+    };
+
+    var getServicePortForRoute = function(targetPort, service) {
+      return _.find(service.spec.ports, function(servicePort) {
+        if (isPortNamed(targetPort)) {
+          // When using a named port in the route target port, it refers to the service port.
+          return servicePort.name === targetPort;
+        }
+
+        // Otherwise it refers to the container port (the service target port).
+        return servicePort.targetPort === targetPort;
```

Comment:
servicePort.targetPort can be a string, in which case we won't be able to correlate the route targetPort, right? that's fine, just making sure I understand